### PR TITLE
fix: adding map for marketo known values and javascript known values

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

While running sync for a customer and we have come accross invalid id error for a huge number of events.

While debugging we found out that the data delivery was happening seemlessly but we have issue in reporting the failure and warning events.

while returning errors or warnings, marketo sends back all the fields of the erroneous events, along with its values and reason for failure, but no job ids are sent back

This the reason, we match all the events we already sent ( field-value ) to marketo with the response we get back from marketo.

The response itself can be clumsy and not so readable at various times. Upon research, we have seen that, if the data type of a field inside Marketo ( that we are making sure to get by sending one axios get call )  does not match with the data type that users send, it sends back a warning. That is why, we are checking, if data types are matching or not.

Now javascript understands 5 types of variables types:

- number
- string
- boolean
- undefined
- object

we had considered any other data types of data to be of type `string`. 

For example : marketo has `email` as a datatype, which is not easy for us to track. But for any email, the `typeOf` function returns string.

But we missed the logic that even if we consider email as `string`, the initial data type from marketo field is still `email`.

That is why this PR contains a suitable mapping for marketo known and javascript known data types

## What is the related Linear task?

Resolves INT-1446

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

**Reason**:

For example : marketo has `email` as a datatype, which is not easy for us to track. But for any email, the `typeOf` function returns string.

But we missed the logic that even if we consider email as `string`, the initial data type from marketo field is still `email`.

**Changes** :

We added a map which we convert marketo specific data types to javascript specific data types.
Example : 
email --> string
integer --> number 

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
       "name": "Launch TS",
       "program": "${workspaceFolder}/dist/src/index.js",
       "outFiles": ["${workspaceFolder}/dist/src/**/*.js"],
-      "runtimeExecutable": "/usr/local/bin/node"
```
Comment (yashasvibajpai) [inline .vscode/launch.json:22]: This shouldn't change right? @shrouti1507

Comment (shrouti1507) [reply .vscode/launch.json:22]: yes reverting

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3037/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3037/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3037/test-report.html

**Review by shrouti1507 — COMMENTED** [review]
@coderabbitai review

File: src/v0/destinations/marketo_bulk_upload/marketo_bulk_upload.util.test.js [hunk 2]
```diff
@@ -353,3 +354,187 @@ describe('getAccessToken', () => {
...
+
+    const result = checkEventStatusViaSchemaMatching(event, fieldSchemaMapping);
+
+    expect(result).toEqual({
+      job1: 'invalid abc',
+    });
+  });
+
+  // The function correctly handles events with null values.
+  it('should correctly handle events with null values', () => {
```
Comment (utsabc) [inline src/v0/destinations/marketo_bulk_upload/marketo_bulk_upload.util.test.js:513]: Make the test case description more elaborative, talk about what is the expectation.

Example

```
it('should correctly handle events with null values returning the fields having null values as invalid',
```

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3037/test-report.html

File: src/v0/destinations/marketo_bulk_upload/marketo_bulk_upload.util.test.js (hunk 2 above)
Comment (shrouti1507) [reply src/v0/destinations/marketo_bulk_upload/marketo_bulk_upload.util.test.js:513]: addressed

**Review by utsabc — DISMISSED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3037/test-report.html

**Review by utsabc — APPROVED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3037/test-report.html

File: src/v0/destinations/marketo_bulk_upload/util.js [hunk 3]
```diff
@@ -401,16 +402,11 @@ const checkEventStatusViaSchemaMatching = (event, fieldMap) => {
...
       const actualDataType = typeof paramValue;
 
-      // If expectedDataType is not one of the primitive data types, treat it as a string
-      if (!['string', 'number', 'boolean', 'undefined'].includes(expectedDataType)) {
-        expectedDataType = 'string';
-      }
-
       if (!mismatchedFields[job_id] && actualDataType !== expectedDataType) {
-        mismatchedFields[job_id] = `invalid ${paramName}`;
+        mismatchedFields[job_id] = `invalid ${paramName} for value ${paramValue} `;
```
Comment (chandumlg) [inline src/v0/destinations/marketo_bulk_upload/util.js:409]: Remove `for value ${paramValue}`

**Review by chandumlg — DISMISSED** [review]

**Review by chandumlg — APPROVED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3037/test-report.html

File: src/v0/destinations/marketo_bulk_upload/util.js (hunk 3 above)
Comment (shrouti1507) [reply src/v0/destinations/marketo_bulk_upload/util.js]: addressed

**Review by utsabc — APPROVED** [review]

**Pull request merged**